src/SDK/Metrics/MetricRegistry/MetricRegistry.php

```diff
@@ -143,6 +148,12 @@ public function collectAndPush(iterable $streamIds): void
                 $aggregator = $this->asynchronousAggregatorFactories[$streamId]->create();
 
                 $instrumentId = $this->streamToInstrument[$streamId];
+                if (
+                    array_key_exists($instrumentId, $this->instruments)
+                    && $this->instruments[$instrumentId]->meter?->isEnabled() === false
```

Comment:
Adjusted MetricReader to not export when source->instrument->meter is not enabled. I've left the above code in place for now, since it also acts to stop observer callbacks from being called (which seems like a performance win, if those callbacks were computationally expensive) - happy to revisit that, though!

Reply:
> since it also acts to stop observer callbacks from being called

We should also skip collecting synchronous aggregators which could be done by moving the `$this->instruments[$instrumentId]->meter->isEnabled()` check to the beginning of the for-loop and populating `$this->instruments` in `::registerSynchronousStream()`/`::registerAsynchronousStream()` instead of `::registerCallback()` (could alternatively be done in `ExportingReader::doCollect()` by removing disabled meters from the stream ids that are passed to `::collectAndPush()`, but keeping it in `MetricRegistry` seems easier with the current setup). But...

> We could also remove the underlying metric streams but this would require more effort.

This seems like the only way to implement it properly, otherwise we will export incorrect data when instrumentation scopes are reenabled.
```php
# w/ temporality=cumulative
$c = $meterProvider->getMeter('test')->createCounter('c');

# t0
$meterConfig->setDisabled(false);
$c->add(1);

# t1
$meterConfig->setDisabled(true);
$c->add(1);

# t2, {sum=1, startTimestamp=t2}; must not export {sum=2, startTimestamp=t0}
$meterConfig->setDisabled(false);
$c->add(1);
$meterProvider->forceFlush();
```

Very rough implementation idea <sub>(alternatively see [implementation with slightly different metrics SDK](https://github.com/Nevay/otel-sdk-core/commit/639cc1521b49b22a96cb25ff2be18e30b7e09541#diff-d13ea4c47a008d26ff07dc3b24ac010d3a0fe63c40c081f417d93a4362a28613))</sub>:
- on disable: remove streams for all instrumentation scope instruments (within `Meter`: `$this->instruments->writers` and `->observers`); logic similar to stalenesshandler callbacks in `Meter` and `ExportingReader` which already remove streams
- on enable: recreate streams for all instrumentation scope instruments by calling `MetricFactory::createSynchronousWriter()`/`::createAsynchronousObserver()`
- `Meter::createSynchronousWriter()`/`::createAsynchronousObserver()` must not create streams if the instrumentation scope is disabled

Reply:
I haven't been able to get this to work, but have left a todo and a skipped integration.